### PR TITLE
Add release guide

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

This PR adds release guide for this project. I referred to the release guide in scheduler-plugins.
https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/RELEASE.md

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,9 +1,42 @@
...
-The Kubernetes Template Project is released on an as-needed basis. The process is as follows:
+## Semi-automatic
 
-1. An issue is proposing a new release with a changelog since the last release
-1. All [OWNERS](OWNERS) must LGTM this release
-1. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
-1. The release issue is closed
-1. An announcement email is sent to `kubernetes-dev@googlegroups.com` with the subject `[ANNOUNCE] kubernetes-template-project $VERSION is released`
+1. Make sure your repo is clean by git's standards
+2. Create a release branch `git checkout -b release-1.19` (not required for patch releases)
```
Comment (alculquicondor) [inline RELEASE.md:6]: This should refer to the OWNERS, as the original doc stated. Very few people have rights to do this, AFAIK.

File: RELEASE.md [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
+8. Email `kubernetes-sig-scheduling@googlegroups.com` to announce the release
+
+## Manual
+
+1. Make sure your repo is clean by git's standards
+2. Create a release branch `git checkout -b release-1.19` (not required for patch releases)
+3. Push the release branch to the kube-scheduler-simulator repo and ensure branch protection is enabled (not required for patch releases)
+4. Tag the repository from the `master` branch (from the `release-1.19` branch for a patch release) and push the tag `VERSION=v0.19.0 git tag -m $VERSION $VERSION; git push origin $VERSION`
+5. Checkout the tag you just created and make sure your repo is clean by git's standards `git checkout $VERSION`
+6. Build and push the container image to the staging registry `RELEASE_VERSION=$VERSION make push-release-image`
```
Comment (alculquicondor) [inline RELEASE.md:21]: Who has permission to do this?

Why not only document the "semi-automatic" way?

File: RELEASE.md [hunk 3]
```diff
@@ -1,9 +1,42 @@
-# Release Process
+# Release Guide
 
-The Kubernetes Template Project is released on an as-needed basis. The process is as follows:
+## Semi-automatic
 
-1. An issue is proposing a new release with a changelog since the last release
```
Comment (Huang-Wei) [inline RELEASE.md:5]: Usually we have one issue showing the todos/progress of a release, so let's keep this step. [An example](https://github.com/kubernetes-sigs/scheduler-plugins/issues/330).

File: RELEASE.md (hunk 2 above)
Comment (Huang-Wei) [reply RELEASE.md:21]: The takeaway of this section is that some forked repo can follow this to release and push images to their own image registry, but maybe not that useful. So I'd like to either remove the whole section or reword it to fit the purpose of forked users.

Comment (sanposhiho) [reply RELEASE.md:21]: OK. I'll delete it.

**Comment by sanposhiho** [comment]
@alculquicondor @Huang-Wei 
Thanks for reviewing. I addressed all your reviews.

File: RELEASE.md [hunk 4]
```diff
@@ -1,9 +1,29 @@
...
-The Kubernetes Template Project is released on an as-needed basis. The process is as follows:
+# Release Guide
 
 1. An issue is proposing a new release with a changelog since the last release
-1. All [OWNERS](OWNERS) must LGTM this release
-1. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
-1. The release issue is closed
-1. An announcement email is sent to `kubernetes-dev@googlegroups.com` with the subject `[ANNOUNCE] kubernetes-template-project $VERSION is released`
+2. Make sure your repo is clean by git's standards
+3. An [OWNERS](OWNERS) creates a release branch `git checkout -b release-1.19` (not required for patch releases)
```
Comment (alculquicondor) [inline RELEASE.md:5]:
```suggestion
3. An [OWNER](OWNERS) creates a release branch `git checkout -b release-1.19` (not required for patch releases)
```

File: RELEASE.md [hunk 5]
```diff
@@ -1,9 +1,29 @@
...
+4. Add the prow-job settings for the new release branch.
```
Comment (alculquicondor) [inline RELEASE.md:6]: where?

File: RELEASE.md [hunk 6]
```diff
@@ -1,9 +1,29 @@
...
 1. An issue is proposing a new release with a changelog since the last release
-1. All [OWNERS](OWNERS) must LGTM this release
-1. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
-1. The release issue is closed
-1. An announcement email is sent to `kubernetes-dev@googlegroups.com` with the subject `[ANNOUNCE] kubernetes-template-project $VERSION is released`
+2. Make sure your repo is clean by git's standards
+3. An [OWNERS](OWNERS) creates a release branch `git checkout -b release-1.19` (not required for patch releases)
+4. Add the prow-job settings for the new release branch. 
+5. Push the release branch to the kube-scheduler-simulator repo and ensure branch protection is enabled (not required for patch releases)
+6. Tag the repository from the `master` branch (from the `release-1.19` branch for a patch release) and push the tag `VERSION=v0.19.0 git tag -m $VERSION $VERSION; git push origin $VERSION`
```
Comment (alculquicondor) [inline RELEASE.md:8]: Shouldn't you tag before branching out?

File: RELEASE.md (hunk 5 above)
Comment (sanposhiho) [reply RELEASE.md]: added the link to test-infra

**Comment by sanposhiho** [comment]
@Huang-Wei @alculquicondor 
Sorry for the late, I completely forgot about this PR.
I updated the PR to address your reviews, please take a look at this again 🙏

File: RELEASE.md [hunk 7]
```diff
@@ -1,9 +1,29 @@
...
 
 1. An issue is proposing a new release with a changelog since the last release
-1. All [OWNERS](OWNERS) must LGTM this release
-1. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
-1. The release issue is closed
-1. An announcement email is sent to `kubernetes-dev@googlegroups.com` with the subject `[ANNOUNCE] kubernetes-template-project $VERSION is released`
+2. Make sure your repo is clean by git's standards
+3. Tag the repository from the `master` branch (from the `release-1.19` branch for a patch release) and push the tag `VERSION=v0.19.0 git tag -m $VERSION $VERSION; git push origin $VERSION`
+4. An [OWNER](OWNERS) creates a release branch `git checkout -b release-1.19` (not required for patch releases)
+5. Add the prow-job settings for the new release branch [here](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs/kube-scheduler-simulator).
```
Comment (196Ikuchil) [inline RELEASE.md:7]: small nit:
```suggestion
5. Add the prow-job settings for the new release branch [here](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs/kube-scheduler-simulator)
```
